### PR TITLE
Boost robustness

### PR DESCRIPTION
**Pull request by jessetilro**
Just preparing the PR and checking build status. Not ready yet, more commits are coming!

### CODE REVIEW

**Comment by dmalarev** [comment]
Lgtm

File: app/src/main/java/nl/tudelft/pixelperfect/pixelperfect/location/LocationArmoryActivity.java [hunk 1]
```diff
@@ -28,19 +30,35 @@
...
+     * @param item the item in the action bar.
+     * @return states whether a function was executed.
+     */
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        switch (item.getItemId()) {
+            case android.R.id.home:
+                onBackPressed();
+                return true;
+        }
```
Comment (wouterz) [inline app/src/main/java/nl/tudelft/pixelperfect/pixelperfect/location/LocationArmoryActivity.java:51]: switch with one case? feel free to leave it but not that pretty

File: app/src/main/java/nl/tudelft/pixelperfect/pixelperfect/location/LocationEngineroomActivity.java [hunk 2]
```diff
@@ -29,19 +31,35 @@
...
+     *
+     * @param item the item in the action bar.
+     * @return states whether a function was executed.
+     */
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        switch (item.getItemId()) {
+            case android.R.id.home:
+                onBackPressed();
+                return true;
```
Comment (wouterz) [inline app/src/main/java/nl/tudelft/pixelperfect/pixelperfect/location/LocationEngineroomActivity.java:51]: again switch with one case

File: app/src/main/java/nl/tudelft/pixelperfect/pixelperfect/location/LocationLabActivity.java [hunk 3]
```diff
@@ -27,19 +29,35 @@
...
+     * itself.
+     *
+     * @param item the item in the action bar.
+     * @return states whether a function was executed.
+     */
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        switch (item.getItemId()) {
+            case android.R.id.home:
+                onBackPressed();
```
Comment (wouterz) [inline app/src/main/java/nl/tudelft/pixelperfect/pixelperfect/location/LocationLabActivity.java:48]: same

**Comment by wouterz** [comment]
dammn jesse super netjes weer!

File: app/src/main/java/nl/tudelft/pixelperfect/pixelperfect/location/LocationArmoryActivity.java (hunk 1 above)
Comment (jessetilro) [inline app/src/main/java/nl/tudelft/pixelperfect/pixelperfect/location/LocationArmoryActivity.java:51]: This was already done this way in other classes, just copy pasted it.

File: app/src/main/java/nl/tudelft/pixelperfect/pixelperfect/location/LocationEngineroomActivity.java (hunk 2 above)
Comment (jessetilro) [inline app/src/main/java/nl/tudelft/pixelperfect/pixelperfect/location/LocationEngineroomActivity.java:51]: This was already done this way in other classes, just copy pasted it.

File: app/src/main/java/nl/tudelft/pixelperfect/pixelperfect/location/LocationLabActivity.java (hunk 3 above)
Comment (jessetilro) [inline app/src/main/java/nl/tudelft/pixelperfect/pixelperfect/location/LocationLabActivity.java:48]: This was already done this way in other classes, just copy pasted it.

**Comment by jessetilro** [comment]
Completed last fix just a minute ago. Ready to merge if Travis agrees.

**Pull request merged**